# Hypothesis Testing Exercise

## 1. Cutlets

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [2]:
df_cutlets=pd.read_csv('Cutlets.csv')
df_cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

Assuming:

* Null hyposthesis as Ho: μ1 = μ2 (There is no difference in diameters of cutlets between two units) 
* Alternate hypothesis as Ha: μ1 ≠ μ2 (There is significant difference in diameters of cutlets between two units) 

###### 2 Sample 2 Tail test applicable

In [3]:
unit_A=pd.Series(df_cutlets.iloc[:,0])
unit_B=pd.Series(df_cutlets.iloc[:,1])

In [4]:
p_value=stats.ttest_ind(unit_A,unit_B)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

### Conclusion:

As (p_value=0.4722) > (α = 0.05) 
* Accept Null Hypothesis i.e. μ1 = μ2 ( Thus, there is no difference in diameters of cutlets between two units)

## 2. LabTAT

In [5]:
df_lab=pd.read_csv('LabTAT.csv')
df_lab.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

#### Anova ftest statistics: Analysis of varaince between more than 2 samples or columns

Assume 
* Null Hypothesis Ho as No Varaince: All samples TAT population means are same
* Alternate Hypothesis Ha as It has Variance: Atleast one sample TAT population mean is different 

In [6]:
p_value=stats.f_oneway(df_lab.iloc[:,0],df_lab.iloc[:,1],df_lab.iloc[:,2],df_lab.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

### Conclusion:

As (p-value=0) < (α = 0.05)
* Reject Null Hypothesis i.e. Atleast one sample TAT population mean is different 
* Thus there is variance or difference in average Turn Around Time (TAT) of reports of the laboratories on their preferred list.

## 3. Buyer Ratio

In [7]:
df_br=pd.read_csv('BuyerRatio.csv')
df_br

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [8]:
data=np.array([[50,142,131,70],[435,1523,1356,750]])
data

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

#### Assume 

* Null Hypothesis as Ho: Independence of categorical variables (male-female buyer rations are similar across regions (does not vary and are not related)

* Alternate Hypothesis as Ha: Dependence of categorical variables (male-female buyer rations are NOT similar across regions (does vary and somewhat/significantly related)

In [9]:
from scipy.stats import chi2_contingency

In [10]:
chi2_contingency(data)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

### Conclusion:

As (p-value = 0.6603) > (α = 0.05)
* Accept the Null Hypothesis i.e. Independence of categorical variables 
* Thus, male-female buyer rations are similar across regions and are not related

## 4. Customer Order Form 

In [11]:
df=pd.read_csv('Costomer+OrderForm.csv')
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [12]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [13]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [14]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [15]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [16]:
data=np.array([[271,267,269,280],[29,33,31,20]])
data

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

#### Assume 

* Null Hypothesis as Ho: Independence of categorical variables (customer order forms defective %  does not varies by centre)
* Alternative hypothesis as Ha: Dependence of categorical variables (customer order forms defective %  varies by centre)

In [17]:
chi2_contingency(data)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

### Conclusion: 

As (p_value = 0.2771) > (α = 0.05)
* Accept Null Hypthesis i.e. Independence of categorical variables 
*Thus, customer order forms defective %  does not varies by centre